### PR TITLE
#1400. Add tests for external members, covariant parameters, nullable supertypes

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Review by eernstg — COMMENTED** [review]
Looks good! Just one thing, I think we need to make an exception for dart2js. Also, we should have some `static` members along with the given instance members.

File: LanguageFeatures/Extension-types/static_analysis_extension_types_A28_t01.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is not an error for an extension type member to have the
+/// modifier external
+///
+/// @description Checks that it is not an error for an extension type member to
+/// have the modifier `external`
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Extension-types/static_analysis_extension_types_A28_t01.dart:9]: Vary good! Thank you for pointing! Forgot about it. We skip such tests in a `.status` files with the comment `...: SkipByDesign # Non-JS-interop external members are not supported`. I'm going to do a new roll and update status files accordingly

**Comment by sgrekhov** [comment]
Added tests for static members and some more

File: LanguageFeatures/Extension-types/static_analysis_extension_types_A28_t01.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Extension-types/static_analysis_extension_types_A28_t01.dart:9]: OK, great!

File: LanguageFeatures/Extension-types/syntax_A10_t02.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+/// AnExtensionType.myStaticMethod(42)
+///
+/// @description Checks that it is a compile-time error if an extension type
+/// declares a static operator
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+extension type ET(int id) {
+  static int operator +(int other) => id + other;
```
Comment (eernstg) [inline LanguageFeatures/Extension-types/syntax_A10_t02.dart:33]: Good catch! Didn't think about that.

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**